sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/SearchServiceAsyncClient.java

```diff
@@ -117,36 +117,38 @@ public String getEndpoint() {
     /**
      * Creates a new Azure Cognitive Search data source or updates a data source if it already exists.
      *
-     * @param dataSource the definition of the data source to create or update
+     * @param dataSource The definition of the {@link DataSource} to create or update.
      * @return the data source that was created or updated.
      */
     public Mono<DataSource> createOrUpdateDataSource(DataSource dataSource) {
-        return createOrUpdateDataSourceWithResponse(dataSource, null, null).map(Response::getValue);
+        return createOrUpdateDataSourceWithResponse(dataSource, false, null).map(Response::getValue);
     }
 
     /**
      * Creates a new Azure Cognitive Search data source or updates a data source if it already exists.
      *
-     * @param dataSource The definition of the data source to create or update.
-     * @param accessCondition the condition where the operation will be performed if the ETag on the server matches or
-     * doesn't match specified values
+     * @param dataSource The definition of the {@link DataSource} to create or update.
+     * @param onlyIfUnchanged boolean to indicate whether to check etag before perform the operation. True means
```

Comment:
The documentation shouldn't contain implementation details.
`@param onlyIfUnchanged {@code true} to update if the {@code dataSource} is the same as the current service value. {@code false} to always update existing value.`

Reply:
Same with any other instances of this.